src/gluonts/distribution/gamma.py

```diff
@@ -0,0 +1,144 @@
+from .distribution_output import DistributionOutput
+
+
+class Gamma(Distribution):
+    r"""
+    Gamma distribution.
+
+    Parameters
+    ----------
+    alpha
+        Tensor containing the shape parameters, of shape `(*batch_shape, *event_shape)`.
+    beta
+        Tensor containing the rate parameters, of shape `(*batch_shape, *event_shape)`.
+    F
+    """
+
+    is_reparameterizable = True
+
+    @validated()
+    def __init__(self, alpha: Tensor, beta: Tensor, F=None) -> None:
+        self.alpha = alpha
+        self.beta = beta
+        self.F = (
+            F if F else getF(alpha)
+        )  # assuming alpha and beta of same type
+
+    @property
+    def batch_shape(self) -> Tuple:
+        return self.alpha.shape
+
+    @property
+    def event_shape(self) -> Tuple:
+        return ()
+
+    @property
+    def event_dim(self) -> int:
+        return 0
+
+    def log_prob(self, x: Tensor) -> Tensor:
+        F = self.F
+        alpha, beta = self.alpha, self.beta
+
+        return (
+            alpha * F.log(beta)
+            - F.gammaln(alpha)
+            + (alpha - 1) * F.log(x)
+            - beta * x
+        )
+
+    @property
+    def mean(self) -> Tensor:
+        return self.alpha / self.beta
+
+    @property
+    def stddev(self) -> Tensor:
+        return self.F.sqrt(self.alpha) / self.beta
+
+    def sample(
+        self, num_samples: Optional[int] = None, dtype=np.float32
+    ) -> Tensor:
+        return _sample_multiple(
+            partial(self.F.sample_gamma, dtype=dtype),
+            alpha=self.alpha,
+            beta=1.0 / self.beta,
+            num_samples=num_samples,
+        )
+
+    def sample_rep(
+        self, num_samples: Optional[int] = None, dtype=np.float32
+    ) -> Tensor:
+        def s(alpha: Tensor, beta: Tensor) -> Tensor:
+            return self.F.sample_gamma(alpha=alpha, beta=beta, dtype=dtype)
+
+        return _sample_multiple(
+            s,
+            alpha=self.alpha,
+            beta=1.0 / self.beta,
+            num_samples=num_samples,  # because MXNet doesnt make sense
+        )
+
+    @property
+    def args(self) -> List:
+        return [self.alpha, self.beta]
+
+
+class GammaOutput(DistributionOutput):
+    args_dim: Dict[str, int] = {"alpha": 1, "beta": 1}
+    distr_cls: type = Gamma
+
+    @classmethod
+    def domain_map(cls, F, alpha, beta):
+        r"""
+        Maps raw tensors to valid arguments for constructing a Gamma
+        distribution.
+
+        Parameters
+        ----------
+        F
+        alpha
+            Tensor of shape `(*batch_shape, 1)`
+        beta
+            Tensor of shape `(*batch_shape, 1)`
+
+        Returns
+        -------
+        Tuple[Tensor, Tensor]
+            Two squeezed tensors, of shape `(*batch_shape)`: both have entries mapped to the
+            positive orthant.
+        """
+        alpha = softplus(F, alpha)
+        beta = softplus(F, beta)
```

Comment:
These values can get arbitrarily close to zero, which can create problems when computing the `log_prob`. We had similar issues with `NegativeBinomial`, and there we add some small constant to make sure we're away from zero: https://github.com/awslabs/gluon-ts/blob/4bc1f36a076ba44f12568036221c75506b0cd5b4/src/gluonts/distribution/neg_binomial.py#L112-L113

We have a test for `NegativeBinomial` that triggers that problem, maybe you can check if something similar occurs here: https://github.com/awslabs/gluon-ts/blob/master/test/distribution/test_negative_binomial.py

Reply:
You were right, the test failed for the Gamma Distribution as well.
We probably should do the tests for the other cases of using `softplus`.